src/test/java/com/google/devtools/build/lib/remote/RemoteServerCapabilitiesTest.java

```diff
@@ -103,6 +105,10 @@ public static void afterEverything() {
       assertThat(meta.getToolDetails().getToolName()).isEqualTo("bazel");
       assertThat(meta.getToolDetails().getToolVersion())
           .isEqualTo(BlazeVersionInfo.instance().getVersion());
+      assertThat(headers.get(Metadata.Key.of("Key1", Metadata.ASCII_STRING_MARSHALLER)))
```

Comment:
Having checks in the common case means that any future test case is forced to also test this by setting up custom remoteOptions. That seems like a bad direction, generally speaking.

Reply:
Right, I'll create a new test case for it.